### ISSUE TITLE
NVME vs M.2 SATA

### ISSUE THREAD

**Issue by applesjgtl** [syscl/XPS9350-macOS#20]
I have a 512GB NVME laptop that had no problem finding the target install disk but the 256GB M.2 SATA laptop can't find the install disk. I'm guessing you have the NVME KEXT from RehabMan, the question is, why is it that my drive isn't showing up on the M.2 computer?

**Comment by zeusk** [comment]
Did you set SATA to AHCI in UEFI?

**Comment by applesjgtl** [comment]
Of course, yes. I am an experienced Hackintosher. I also reset UEFI and changed the settings again, just to verify. I know that the clover folder from maxecloo will result in a functional installer so obviously AHCI is enabled and functional.


On Tue, Dec 20, 2016 at 2:04am, Shantanu Gupta < notifications@github.com [notifications@github.com] > wrote:
Did you set SATA to AHCI in UEFI?

**Comment by syscl** [comment]
@applesjgtl No, I never use RehabMan's NVME driver...

Note: Do NOT use any acpi tables other than from your computer. This is the reason why I start this project. In my project, deploy.sh will dynamically patch all the ACPI tables from your computer, which is the most stable way...

And, you have to uncheck dropOEM(dropOEM = false) to let system identify your drive.

syscl

**Comment by gctseng** [comment]
@syscl @applesjgtl I'm in the process of reinstalling OS X, as I mentioned in my other thread.  The CLOVER folder in the USB drive is also from the current commit.  I noticed when I tried to boot in verbose mode, it wouldn't see my 512GB NVME.  I tried using Macbook9,1 in the config.plist (I copied _only_ the SMBIOS section only to the current config's config.plist) and the installer just rebooted when I clicked on macOS Sierra (a KP?).  Switched the SMBIOS in the config.plist back to the one in the current commit and it seems like it's installing now.

**Comment by gctseng** [comment]
LOL.  FML.  My laptop freaked out.  LCD panel/video error code, won't boot anymore.  I was changing the boot option value in the boot selection screen.  See below.  I have to either return it to Costco or send it into Service Center. 

I had a feeling there was something wrong ever since I got it.

![img_4875](https://cloud.githubusercontent.com/assets/7102506/21378878/814d1334-c6fe-11e6-8f8d-1daab3ad1db9.JPG)
![img_4869](https://cloud.githubusercontent.com/assets/7102506/21378879/814f6954-c6fe-11e6-9d7b-410e17232194.JPG)

**Comment by applesjgtl** [comment]
Reinstalled macOS, totally stock 10.12.2 - using your CLOVER folder and after running latest deploy.sh, still not booting.

**Comment by applesjgtl** [comment]
![img_20161226_181019](https://cloud.githubusercontent.com/assets/2176220/21488603/b794b0f4-cb96-11e6-9290-0b0a1f7e54bf.jpg)

**Comment by syscl** [comment]
I know why, because you have non-iris model, you have to do the following:

- In Clover: set Inject Intel = False
- Boot macOS

When u finish install, Inject Intel = False to boot the sys, run deploy.sh to finish the installation...


syscl

**Comment by applesjgtl** [comment]
deploy.sh was already run, still stuck on boot with false inject Intel.

**Comment by applesjgtl** [comment]
![img_20161226_182032](https://cloud.githubusercontent.com/assets/2176220/21488762/2b2c6998-cb98-11e6-8ba1-04f20c2bb64b.jpg)
![14828053207941749432918](https://cloud.githubusercontent.com/assets/2176220/21488766/36967e86-cb98-11e6-8a54-2c5047f13392.jpg)

**Comment by syscl** [comment]
Do you remove all of your a picture tables?

What if-platform-id you have under Clover GUI?

**Comment by applesjgtl** [comment]
via Newton Mail [https://cloudmagic.com/k/d/mailapp?ct=pa&cv=9.3.39&pv=7.0&source=email_footer_2] On Mon, Dec 26, 2016 at 6:23pm, Zhou Yating < notifications@github.com [notifications@github.com] > wrote:
Do you remove all of your a picture tables?

What if-platform-id you have under Clover GUI?

**Comment by applesjgtl** [comment]
![cm-img-20161226_182459](https://cloud.githubusercontent.com/assets/2176220/21488829/d2845d40-cb98-11e6-93c2-b0b1b80b1cde.jpg)

**Comment by applesjgtl** [comment]
![1482805772856948448105](https://cloud.githubusercontent.com/assets/2176220/21488859/488143f0-cb99-11e6-9024-9e24b4da205b.jpg)

**Comment by applesjgtl** [comment]
I don't know why, but your default theme renders text poorly on these hidpi displays. Maybe you should switch to embedded for readability.

**Comment by syscl** [comment]
ig-platform-id is incorrect, set to 0x19160000

It's weird that I have already considered this problem, please give me the full result of the deploy.sh

syscl

**Comment by applesjgtl** [comment]
I already set 0x19160000 manually and booting still does not occur; hangs in same place.

**Comment by applesjgtl** [comment]
deploy.sh does not have any errors echoed, is there anywhere else that it puts a log?

**Comment by syscl** [comment]
@applesjgtl Oh, I see, go to ACPI under Clover and set DSDT.aml to DSDTxx.aml

syscl

**Comment by applesjgtl** [comment]
Done. Still hanging.

**Comment by applesjgtl** [comment]
I can boot successfully (albeit missing functionality like WiFi and sound) with maxecloo's CLOVER folder so if there's anything I can upload to help with debugging let me know.

**Comment by syscl** [comment]
@applesjgtl How about only use his config.plist to replace my Clover folder?

syscl

**Comment by applesjgtl** [comment]
![148280730973062332304](https://cloud.githubusercontent.com/assets/2176220/21489236/e36a20aa-cb9c-11e6-97d3-03acb4a88745.jpg)
Config doesn't fix issue - backtracke mentions displaylink driver.

**Comment by syscl** [comment]
@applesjgtl Oh, how about remove displaylink.kext then boot?

syscl

**Comment by applesjgtl** [comment]
This is not the solution.
http://plugable.com/2010/06/16/workaround-for-mac-os-x-conflicts-between-displaylink-and-usb-ethernet/
it's because I had my USB Ethernet adapter plugged in during boot.

**Comment by syscl** [comment]
@applesjgtl I have no idea what your issue is. If displaylink stuck your boot, you should remove it... 

There's a new BIOS 1.4.12 available, you can try to see if that bios can fix your issue or not.

syscl

**Comment by syscl** [comment]
@applesjgtl BootCamp's font issue is not my problem, new available clover's issue.

syscl

**Comment by applesjgtl** [comment]
DisplayLink error only occurs when USB Ethernet is plugged in. I'm on
latest BIOS.

**Comment by syscl** [comment]
@applesjgtl And I notice that you do not use the latest config.plist

syscl

**Comment by applesjgtl** [comment]
It was latest as of post time - has it been updated since then?


via Newton Mail [https://cloudmagic.com/k/d/mailapp?ct=pa&cv=9.3.39&pv=7.1.1&source=email_footer_2] On Mon, Dec 26, 2016 at 9:52pm, Zhou Yating < notifications@github.com [notifications@github.com] > wrote:
@applesjgtl [https://github.com/applesjgtl] And I notice that you do not use the latest config.plist

syscl

**Comment by syscl** [comment]
The screenshot of your Clover GUI clearly indicate you have MacBook8,1, which is very old version......

syscl

**Comment by applesjgtl** [comment]
The last screenshot is when you asked me to exchange config.plist

**Comment by applesjgtl** [comment]
Running out of ideas.
![img_20170108_170802](https://cloud.githubusercontent.com/assets/2176220/21755232/5d6b43ac-d5c5-11e6-9fdb-2d4cce56d5f1.jpg)
![img_20170108_171014](https://cloud.githubusercontent.com/assets/2176220/21755236/756694c0-d5c5-11e6-8c41-6eca6bb8590d.jpg)

**Comment by syscl** [comment]
@applesjgtl my new returned laptop cannot detect NVMe device now! Please don't hesitate to report to Pike's blog [here](https://pikeralpha.wordpress.com/2017/01/14/does-apple-load-microcode-for-the-igpu/)

Thank you.
syscl

**Comment by applesjgtl** [comment]
I’m glad we’re on the same page with issues here. I still can’t get the i5 working reliably.

via Newton Mail [https://cloudmagic.com/k/d/mailapp?ct=dx&cv=9.3.50&pv=10.12.3&source=email_footer_2]
On Sun, Jan 15, 2017 at 3:17 PM, Zhou Yating <notifications@github.com> wrote:
@applesjgtl [https://github.com/applesjgtl] my new returned laptop cannot detect NVMe device now! Please don't hesitate to report to Pike's blog here [https://pikeralpha.wordpress.com/2017/01/14/does-apple-load-microcode-for-the-igpu/]

Thank you.
syscl

**Comment by syscl** [comment]
@applesjgtl try this patch

```
Name:    IONVMeFamily
find:       f6 c1 10 0f 85 1c 01 00 00
replace: f6 c1 01 0f 85 1c 01 00 00      
```

Good luck!
syscl

**Comment by applesjgtl** [comment]
@syscl has this been added to the official deploy or should I apply it manually? No dice with latest deploy / CLOVER folder.

**Comment by applesjgtl** [comment]
Also, pretty sure my troubles are on a non-NVMe drive.

**Comment by syscl** [comment]
@applesjgtl Add it manually.

syscl
